### DISCUSSION TITLE
firefox: cannot access v4l2loopback device (private-dev)

### DISCUSSION THREAD

**Discussion by gcb** [netblue30/firejail#6344] [Q&A] [answered]
I'm trying to expose a [v4l2loopback](https://github.com/umlaeute/v4l2loopback) device from OBS to firefox.

OBS is kinda weird as it works if I have one device, and fails if I have 2+.

Firefox doesn't work.

If I start both without firejail, everything works.

File `/dev/video9` is the device file I'm using.

Relates to:

* #2723

**Comment by gcb** [comment]
After testing with both applications without firejail, now it also works when using firejail... will debug some more. But apparently, it all works.

**Reply by rusty-snake** [reply, marked as answer]
If `/dev/video9` is created after you started firefox, it will not appear. Add `ignore private-dev` if you need plug 'n' play.

**Comment by gcb** [comment]
Thanks.  It works but required learning one weird thing...

I set up everything before starting firefox, as you said, but then must request access to the camera API using a camera that is NOT a v4l2loopback device. After that request to the device that is v4l2loopback device will work and firefox will show the permission with the video9 Name. If I try that device i know is the video9 as the first one, i never get the permission request. None of this happens without `firejail`.

I will try more with `ignore private-dev`
